### ISSUE TITLE
Bug: False-positive in eol-last-rule

### ISSUE THREAD

**Issue by SimonAdesso** [eslint/eslint#17313]
### Environment

Node version: v16.20.0
npm version: 8.19.4
Local ESLint version: 8.43.0
Global ESLint version: 8.43.0
Operating System: Ubuntu

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>"rules": {
        "eol-last": [
          "error",
          "always"
        ]
      }</summary>

```
{
  "root": true,
  "ignorePatterns": [
    "projects/**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "project": [
          "tsconfig.json"
        ],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "app",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": [
              "app"
            ],
            "style": "kebab-case"
          }
        ]
      }
    },
    {
      "files": [
        "*.html"
      ],
      "extends": [
        "plugin:@angular-eslint/template/recommended"
      ],
      "rules": {
        "semi": [
          "error",
          "always"
        ],
        "eol-last": [
          "error",
          "always"
        ]
      }
    }
  ]
}

```
</details>

```js
import { Observable } from 'rxjs';
import { MessageService } from '../message.service';
import { Component } from '@angular/core';

@Component({
  selector: 'app-component',
  template: '{{ lastMessage$ | async }}'
})
export class MyComponent {

  public lastMessage$: Observable<string> | undefined = this.messageService.getLastMessage$();

  constructor(private messageService: MessageService) {
  }
}

```

### What did you expect to happen?

There should no new line in Angular templates be expected.

### What actually happened?

The rule expects a new-line in the template syntax of the Angular component

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/eslint-eol-last-in-angular-template-jbqqc8?file=src%2Fmain.ts

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Hi, I found a false-positive with `eol-last-rule` while using "plugin:@angular-eslint/template/process-inline-templates" to validate TypeScript in Angular component templates. There should be a possibility to ignore rules for Angular-templates like eol-last-rule.

**Comment by Rec0iL99** [comment]
Hi @SimonAdesso, thanks for the issue. 

> There should no new line in Angular templates be expected.

```js
@Component({
  selector: 'app-component',
  template: '{{ lastMessage$ | async }}'
})
```

I'm not familiar with Angular. Is the above code a Component template in Angular?

**Comment by sam3k** [comment]
Hey @SimonAdesso, I'm getting the following error when I try to run the app:

![Image](https://github.com/eslint/eslint/assets/180152/5fb0beb8-bad7-410e-866f-071992e0ad26)

Am I missing something? Do I need to transpile it somehow?
